src/realm/object-store/CMakeLists.txt

```diff
@@ -116,22 +107,38 @@ if(REALM_ENABLE_SYNC)
         sync/impl/sync_file.hpp)
 
     list(APPEND SOURCES
-        sync/app.cpp
-        sync/app_credentials.cpp
-        sync/app_user.cpp
         sync/app_utils.cpp
         sync/async_open_task.cpp
         sync/generic_network_transport.cpp
         sync/impl/app_metadata.cpp
         sync/impl/sync_file.cpp
         sync/jwt.cpp
-        sync/mongo_client.cpp
-        sync/mongo_collection.cpp
-        sync/mongo_database.cpp
-        sync/push_client.cpp
         sync/sync_manager.cpp
         sync/sync_session.cpp)
 
+    if (REALM_APP_SERVICES)
```

Comment:
auth_request_client, subscribable, app_utils, app_metadata, and generic_network_transport are also app services things.

Reply:
Thanks. I made the changes, except for generic_network_transport because it holds AppError which is part of the SyncUser interface. I think this is fine to just include it as-is, but we could also rename it or take AppError out of the app namespace?

Reply:
Ah yeah, I wasn't quite sure what to do with AppError. generic_network_transport.hpp is lightweight enough that the only reason to do anything with it would be we're worried about it being confusing, and I don't think that's a big problem.